apps/server/src/modules/files-storage/repo/filerecord.repo.ts

```diff
@@ -46,11 +51,50 @@ export class FileRecordRepo extends BaseRepo<FileRecord> {
 		return result;
 	}
 
-	async findBySecurityCheckRequestToken(token: string): Promise<FileRecord> {
+	public async findBySecurityCheckRequestToken(token: string): Promise<FileRecord> {
 		// Must also find expires in future. Please do not add .byExpires().
 		const scope = new FileRecordScope().bySecurityCheckRequestToken(token);
+		const fileRecord = await this.findOne(scope);
 
-		const fileRecord = await this.findOneOrFail(scope);
+		return fileRecord;
+	}
+
+	public async delete(fileRecords: FileRecord[]): Promise<void> {
+		const entities = await this.find(fileRecords);
```

Comment:
First we tried passing a reference instead of an entity to to the entity manager. What was the result of that approach in the end? Doing the find before the remove still feels strange.